### DISCUSSION TITLE
v2.4.0

### DISCUSSION THREAD

**Discussion by kdrag0n** [kdrag0n/safetynet-fix#244] [Announcements]
## Highlights

- **Play Integrity bypass** without breaking device checks or causing other issues
- Disabled use of hardware attestation on Pixel 7 and newer (@anirudhgupta109)

## Other changes

- Updated instructions for newer Android and Magisk versions
- Better debugging for future development

It's taken a while to find way to bypass Play Integrity that doesn't require spoofing the build fingerprint permanently, but I wanted to make sure this module doesn't cause any unnecessary breakage. Just like the original goal of Universal SafetyNet Fix, this minimizes adverse effects by spoofing dynamically at runtime only when necessary. Enjoy!

## Donate

If you found this helpful, please consider supporting development with a **[recurring donation](https://patreon.com/kdrag0n)** for rewards such as early access to updates, exclusive behind-the-scenes development news, and priority support. Alternatively, you can also [buy me a coffee](https://paypal.me/kdrag0ndonate). All support is appreciated :heart: 

<hr /><em>This discussion was created from the release <a href='https://github.com/kdrag0n/safetynet-fix/releases/tag/v2.4.0'>v2.4.0</a>.</em>

**Comment by kotah81** [comment]
delightful

**Comment by dl200010** [comment]
This version actually works on Netflix!!! Thanks so much!

**Comment by Fede2782** [comment]
Amazing work! It's so easy to use, no need anymore to modify props

**Comment by FelemixX** [comment]
Oneplus 7t Oxygen OS 12 MP3 fingerprints does not work :(

**Comment by dchard** [comment]
Works nicely on Xiaomi Mi 11 Lite 5G NE with Android 13. All banking apps work as they should now. With previous version one banking app, and one payment integrated app (for public transport) was randomly crashing.

**Comment by AmirAfsa2006** [comment]
have i can bypass strong integrity? please someone help me

**Reply by Fede2782** [reply]
No. With a unlocked bootloader passing the STRONG INTEGRITY is impossible. The only way is to use the stock rom with locked bootloader

**Reply by fillwithjoy1** [reply]
@Fede2782 Would it be possible to sign the image with your own keys and lock the bootloader, something similar to GrapheneOS? Maybe that could get it to pass STRONG?

**Reply by Fede2782** [reply]
I would be possible only on Google pixel in any case. I don't have idea if that would work, maybe there are restrictions when locked with custom key

**Reply by fillwithjoy1** [reply]
nvrmind, grapheneos has locked bootloader and even that doesn't pass strong

**Reply by Fede2782** [reply]
This doesn't prove anything, grapheneos removed some google specific code and run gms in a sandbox not the actual device. This test should be done with a aosp rom

**Reply by e-t-l** [reply]
If you can figure it out, please update us! I am running rooted GrapheneOS on a Pixel 8 Pro WITH locked bootloader, and it doesn't pass Strong Attestation, maybe because of how they altered the ROM like you said. But that means maybe a magisk module could fix it

**Reply by fillwithjoy1** [reply]
> If you can figure it out, please update us! I am running rooted GrapheneOS on a Pixel 8 Pro WITH locked bootloader, and it doesn't pass Strong Attestation, maybe because of how they altered the ROM like you said. But that means maybe a magisk module could fix it

I don't think you realize how strong integrity works and why you won't be able to bypass it.

**Comment by ibart** [comment]
It used to work great, yesterday wallet stopped me to pay cause of device not fulfilling etc...

Today I tried YASNAC and i failed CTS profile match, out of the blue.

**Comment by eBug** [comment]
on my Pixel XL, SafetyNet Status gave me a CTS profile match 'fail' but after `Clear storage` of Google Play Store, I'm getting a `Passed`
Kudos!

**Reply by zQueal** [reply]
This, and a series of reboots seems to have worked for me.

**Comment by siiidar** [comment]
would somebody kindly let me know how I can use these files for emulator in windows ?

**Comment by SzBenedek2006** [comment]
Hi, sadly my device (Samsung Galaxy A52 5G) doesn't pass any of the checks in Play Integrity API. I kindly ask someone who has any idea, why is this to let me know. Thanks.

![Screenshot_20230129_115328_Play Integrity API Checker.png](https://user-images.githubusercontent.com/95222597/215321623-adf3106f-5c75-4709-8c51-c087ef47df70.png)

![Screenshot_20230129_115353.png](https://user-images.githubusercontent.com/95222597/215321622-ee0483e6-309a-4c5e-a62b-1be577fb6ee3.png)

![Screenshot_20230129_115348.png](https://user-images.githubusercontent.com/95222597/215321625-56bda555-2d78-4bc8-9558-83524d31950d.png)


![Screenshot_20230129_115339.png](https://user-images.githubusercontent.com/95222597/215321624-a57a6728-eed8-42e6-b4f3-87814f874fb1.png)

**Reply by Homi74** [reply]
Enable Enforce DenyList.

Untick on DenyList:
Google Play Service, Google Play Store, Play Integrity API Checker, clear the data of those apps.

**Reply by SzBenedek2006** [reply]
Thanks for your help, now it's better, but still not perfect:

![Screenshot_20230129_183404_Play Integrity API Checker.png](https://user-images.githubusercontent.com/95222597/215345597-8a032d42-61dd-4e15-bc7e-39bd6fe9b4b1.png)

The bigger problem is, that some apps are still detecting root like before.

Hungarian telekom app for example:
![Screenshot_20230129_183420_Telekom.png](https://user-images.githubusercontent.com/95222597/215345624-99033a36-62fe-4c3b-a271-cb76df748d9f.png)

Is there any way to block apps from detecting root?

**Reply by Fede2782** [reply]
Put that app in denylist

**Reply by Homi74** [reply]
Put Hide The Magisk app in settings + put them in blacklist "Hungarian telekom app" etc.

**Reply by SzBenedek2006** [reply]
I tried hide magisk app and DenyListing the apps, but no luck.

**Reply by Homi74** [reply]
Is it this one?
https://play.google.com/store/apps/details?id=hu.telekom.telekomapp&hl=en
If so, it works for me with root.
![Screenshot_20230129-193417](https://user-images.githubusercontent.com/69285725/215348356-e5d3b028-92cf-4ee7-bc40-2874481a5cef.png)

![Screenshot_20230129-192643](https://user-images.githubusercontent.com/69285725/215348198-7c0bc373-7516-41b4-9f25-a935aec0240b.png)

**Reply by SzBenedek2006** [reply]
Yes, that's the app.
It maybe has to do something with One UI. Samsung really locks down their devices. It's almost like Apple.

**Reply by Homi74** [reply]
Have you activated Enforce DenyList?
I can't think of anything else :-(

**Reply by SzBenedek2006** [reply]
Wow the Telekom app is now **working**, and I got passing device integrity and basic integrity.

Sadly the [OTP Bank HU](https://play.google.com/store/apps/details?id=hu.otpbank.mobile) app doesn't work. It opens a website.

Samsung pass detecting root too.

Also I checked again the play integrity checker (with and without enforcing denylist) and it has only basic integrity now, but The [Telekom](https://play.google.com/store/apps/details?id=hu.telekom.telekomapp) app is still **working**.

**Reply by Homi74** [reply]
![Screenshot_20230129-212915](https://user-images.githubusercontent.com/69285725/215354477-9aa5d370-ed0c-4a80-946e-c7a2db08859b.png)
But ... I'm not using v2.4.0, because I couldn't get GooglePay to work with this version, the rest worked without a problem, now I have v2.3.1-MOD_2.1 and everything works as it should.

viz:
https://forum.xda-developers.com/t/magisk-module-universal-safetynet-fix-2-4-0.4217823/page-91#post-87198517

**Reply by SzBenedek2006** [reply]
Thanks, I will try it.

**Reply by eBug** [reply]
Google Pay /Wallet is working here with 2.4.0
What I did was:

1. Change Magisk name (some apps check if the app is installed by name)
2. Use zygisk
3. Disable `Enforce DenyList`
4. Add banking apps to `DenyList`  + Google Wallet (Google Play services not included)
5. Clear storage for Google Play Store
6. Restart phone

**Reply by Homi74** [reply]
If Disable Enforce DenyList, banking apps detect root and don't work.
![Screenshot_20230129-223910](https://user-images.githubusercontent.com/69285725/215357112-a957a6fd-0d36-4617-a2f2-132f874849c7.png)
I've already tried everything.

**Reply by eddysapata** [reply]
There is an app AIR frozen it is an very old app but it works perfectly with my banking app(HDFC). 
Hide magisk from setting inside magisk and the add magisk in airfrozen. 
@Homi74 
P.s I am using pixel 7 pro

**Reply by DistroAgnostic** [reply]
> There is an app AIR frozen it is an very old app but it works perfectly with my banking app(HDFC). 
> Hide magisk from setting inside magisk and the add magisk in airfrozen. 
> @Homi74 
> P.s I am using pixel 7 pro

I tried your method but hdfc is not working for me. Please help.

**Comment by Ylatinol** [comment]
Como areglar el error de camara en A52S 5G SM-A528B/DS en android 13

**Comment by sawork** [comment]
Google Pixel 5 + Lineage OS 20 + Magisk 25206 + Safety Net 2.4: Doesn't pass verification CTS Profile.
Google Pixel 5 + Lineage OS 20 + Magisk 25206 + Safety Net 2.3.1: Works great.

**Reply by GROOVYJOSHCLARK** [reply]
> 2.3.1

Holy crap, you are DEAD RIGHT! My pixel 4XL, running official LOS 20 with using SafetyNetFix 2.40, CTS profile mismatch, FAILS! HOWEVER, rolling BACK to the PREVIOUS SNF to 2.3.1, reboot, wipe Play Store data (for NF and to become certified), bingo! CTS profile pass! Thanks for reporting this, and for others, if you FAIL CTS profile, use SNF 2.3.1!!! Netflix pops up in Play Store, I am certified, and SN passes!!!!!

**Comment by Moalamir** [comment]
Can't download

**Comment by Homi74** [comment]
> Can't download 
No problem downloading.

https://github.com/kdrag0n/safetynet-fix/releases/download/v2.4.0/safetynet-fix-v2.4.0.zip

**Comment by Moalamir** [comment]
Thanx

**Comment by BerndLauert1** [comment]
Version 2.4.0 doesn't work for me neither. It continually crashes Google Play Service and i don't pass tests with YASNAC.
A downgrade to 2.3.1 works perfect.

Using Cubot King Kong 3, Android 8.1, Magisk 26.1 + Zygisk

**Comment by husharon1** [comment]
Hey, since the last update 2.4.0 WiFi calling doesn't work. S21 ultra.

**Comment by zQueal** [comment]
v2.4.0 was working there for a while, but after a few days Google Wallet would trigger the security message. I ended up getting v2.4.0-MOD_1.2 by kdrag0n, modded by Displax off [Github](https://github.com/Displax/safetynet-fix/releases/tag/v2.4.0-MOD_1.2) and it's been working for several days without issue.

**Comment by andersonfer100** [comment]
I have a poco m4 pro it has android 13, and it is failing the CTS profile match @kdrag0n Would you help me

**Reply by zQueal** [reply]
Give this a try: https://github.com/Displax/safetynet-fix/releases/tag/v2.4.0-MOD_1.2

**Reply by Displax** [reply]
https://github.com/Displax/safetynet-fix/releases/tag/v2.4.0-MOD_1.3

**Reply by ShlomiPorush** [reply]
Thank you.. this work for me. OP8T android 13

**Comment by LordBli** [comment]
Hello 

I have a Xiaomi 13 ultra and I didn't pass the safetynet check...

Also, regarding the Play Integrity API, I didn't pass the device integrity but I did pass the basic Integrity...


Some advices ?

**Reply by Hot12345** [reply]
https://github.com/Displax/safetynet-fix/releases/tag/v2.4.0-MOD_1.3

**Comment by yacc143** [comment]
Out of curiosity, has anyone figured out how the ELBA Raiffeisen Bank App detects root?

CrDroid 9.8 on Poco F3 here, Magisk 26200, Shamiko 0.7.3, USF 2.4.0.

Playstore, Playservices as usual blocked, ELBA app blocked => "you phone is rooted and unsecure".

Revolut is also on the block list => Revolut does not detect root and works.

(Notice that ELBA does detect root, butit does not refuse to work)

**Reply by crok** [reply]
Freeze the Magisk app with AirFrozen or IceBox - that works with MyRaiffeisen HU.

**Reply by Hot12345** [reply]
Wauw does this really work? It this a knowing issue?

**Comment by LordBli** [comment]
Hi !

I do not know why but I do not pass the device integrity. I'm using safetynet-fix v2.4.0-MOD_1.3.
It was working fine a moment ago.

**Reply by Homi74** [reply]
https://github.com/chiteroman/PlayIntegrityFix/releases/tag/v12.1

**Reply by LordBli** [reply]
Thanks it did work but I cannot use Gpay. Is there any fix ?

**Reply by Homi74** [reply]
try to delete Gpay application data.

**Reply by LordBli** [reply]
I did it also doesn't work...
I'm using a Xiaomi, I found this thread : https://xiaomi.eu/community/threads/google-wallet-stopped-working-device-doesnt-meet-security-requirements.70444/page-2#post-704186

**Reply by Homi74** [reply]
SEEMS TO BE NOT WORKING IN XIAOMI.EU ROMS, WAIT FOR THEIR DEVELOPERS TO FIX IT

**Reply by LordBli** [reply]
yup...

**Reply by ShlomiPorush** [reply]
Using 2.4.0 mod 1. 3 on OnePlus8T and today GPay stop working even I do pass Basic integrity check

**Reply by LordBli** [reply]
Same here ...
I'll try with Zygisk Next ... But Google is messing with us

**Comment by farmerxt** [comment]
worked perfectly over years on galaxy a71 and banking app bank millennium 

now it stopped working ;()

**Comment by avidichard** [comment]
Google Pixel 7 Pro, installed with LineageOS 20 Android version 13. Latest Magisk Module with "Hide Magisk" "Systemless Host" "Zygisk" and enabled "DenyList". Using YASNAC which display's FAIL for CTS profile match using the latest 2.4.0 build. I need my banking apps such as BDO, RCBC and GCash which do not open, opens then closes immediately. These are Philippines apps I can open on one of my older phone with stock ROM no magisk.

**Reply by zQueal** [reply]
This patch no longer works for me at all. Give this one a try with Shamiko, it ended up working for me: https://github.com/chiteroman/PlayIntegrityFix

**Comment by P-Santuy** [comment]
👍

**Comment by Helppmea** [comment]
The application hás an erro and I can't buy the keys to use help me please..

**Reply by fillwithjoy1** [reply]
1. Are you passing BASIC_INTEGRITY?
2. Passing DEVICE_INTEGRITY?
3. Clear Play Store app data completely
4. Retry making that purchase

**Reply by Helppmea** [reply]
I roited It with magisk and ibstalles the camera app, everything was ok, buy aftwr two days when I enter the app It days that the server is down..and the website tô purchase temporary keys to use the application is down..What do I do?
![Screenshot_20240531-114419](https://github.com/kdrag0n/safetynet-fix/assets/171271516/468b0fb8-2e45-4cbd-917c-b8d0e0aef308)

**Reply by Helppmea** [reply]
![Screenshot_20240531-115335](https://github.com/kdrag0n/safetynet-fix/assets/171271516/88014142-6892-46c9-bed6-36e0e7d159ca)

**Reply by Helppmea** [reply]
I installed the virtual camera from the link not from the playstore
